src/modules/Dashboard/features/Applet/Popups/AddParticipantPopup/PublicLinkPopup/PublicLinkPopup.tsx

```diff
@@ -0,0 +1,113 @@
+import { Button } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+
+import { deleteAppletPublicLinkApi, postAppletPublicLinkApi } from 'api';
+import { Modal, Spinner } from 'shared/components';
+import { useAsync } from 'shared/hooks';
+import { StyledBodyLarge, StyledFlexAllCenter, StyledFlexColumn } from 'shared/styles';
+
+import { PublicLinkPopupProps } from './PublicLinkPopup.types';
+
+export const PublicLinkPopup = ({
+  appletId,
+  'data-testid': dataTestId,
+  hasPublicLink = false,
+  onClose,
+  ...otherProps
+}: PublicLinkPopupProps) => {
+  const { t } = useTranslation('app');
+  const {
+    execute: createPublicLink,
+    isLoading: createIsLoading,
+    error: createError,
+  } = useAsync(postAppletPublicLinkApi, (res) => {
+    if (res?.data?.result) {
+      onClose?.(true);
+    }
+  });
+  const {
+    execute: deletePublicLink,
+    isLoading: deleteIsLoading,
+    error: deleteError,
+  } = useAsync(deleteAppletPublicLinkApi, (res) => {
+    if (res?.status === 204) {
+      onClose?.();
+    }
+  });
+  const isLoading = createIsLoading || deleteIsLoading;
+  const hasError = (createError || deleteError) && !isLoading;
+
+  const handleCreatePublicLink = (requireLogin: boolean = false) => {
+    if (appletId) {
+      createPublicLink({ appletId, requireLogin });
+    }
+  };
+
+  const handleDeletePublicLink = () => {
+    if (appletId) {
+      deletePublicLink({ appletId });
+    }
+  };
+
+  return (
+    <Modal
+      data-testid={dataTestId}
+      footer={
+        <StyledFlexAllCenter sx={{ gap: 1.6, width: '100%' }}>
+          {hasPublicLink ? (
+            <Button
+              color="error"
+              data-testId={dataTestId && `${dataTestId}-delete-bttn`}
+              onClick={handleDeletePublicLink}
+              variant="contained"
+            >
+              {t('delete')}
+            </Button>
+          ) : (
+            <>
+              <Button
+                data-testId={dataTestId && `${dataTestId}-no-account-bttn`}
+                disabled={isLoading}
+                onClick={() => {
+                  handleCreatePublicLink(false);
+                }}
+                variant="outlined"
+              >
+                {t('publicLinkCreateAccountNotRequired')}
+              </Button>
+
+              <Button
+                data-testId={dataTestId && `${dataTestId}-account-bttn`}
+                disabled={isLoading}
+                onClick={() => {
+                  handleCreatePublicLink(true);
+                }}
+                variant="outlined"
+              >
+                {t('publicLinkCreateAccountRequired')}
+              </Button>
+            </>
+          )}
+        </StyledFlexAllCenter>
+      }
+      hasActions={false}
+      onClose={() => onClose?.()}
+      title={hasPublicLink ? t('deletePublicLink') : t('publicLinkCreate')}
+      {...otherProps}
+    >
+      <StyledFlexColumn sx={{ gap: 1.6, px: 3.2 }}>
```

Comment:
We have a shared `<StyledModalWrapper>` component that we use to apply standard standard horizontal & vertical padding to the main content of modals. (Ideally, I think such standard paddings should be applied in the `<Modal>` definition itself, however I think there are some exceptions that would require some investigation before applying universally.) For now, you could just wrap your `<StyledFlexColumn>` with `<StyledModalWrapper>` (or apply needed flex styling to it as is done in `AddParticipantPopup`) and remove the `px: 3.2`. Doing so will also make the vertical spacing match Figma.